src/ddmd/expressionsem.d

```diff
@@ -7619,6 +7619,19 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
         {
             if (t2.ty == Tpointer)
             {
+                // https://issues.dlang.org/show_bug.cgi?id=11006
+                Type p1 = t1.nextOf();
```

Comment:
I don't much point here to linking to the bug report. Instead link to the relevant spec
https://dlang.org/spec/expression.html#add_expressions and add the quote: "If both operands are pointers, and the operator is -, the pointers are subtracted and the result is divided by the size of the type pointed to by the operands. It is an error if the pointers point to different types."

Linking to and quoting from the spec should be a good way to clarify what's going on in the implementation.

Reply:
Done.  Thanks!